### PR TITLE
adds the ability to ignore types by selector

### PR DESCRIPTION
**Pull request by rhubarbselleven**
You can add a selector to `ignoredTypes` which will exclude the field from serialise / deserialise if it matches

### CODE REVIEW

File: src/backbone.syphon.js [hunk 1]
```diff
@@ -92,11 +92,14 @@ var getInputElements = function(view, config) {
...
     var reject;
-    var type = getElementType(el);
-    var extractor = config.keyExtractors.get(type);
+    var myType = getElementType(el);
+    var extractor = config.keyExtractors.get(myType);
     var identifier = extractor($(el));
 
-    var foundInIgnored = _.include(config.ignoredTypes, type);
+    var foundInIgnored = _.find(config.ignoredTypes, function(ignoredTypeOrSelector) {
+      return (ignoredTypeOrSelector === myType) || $(el).is(ignoredTypeOrSelector);
```
Comment (samccone) [inline src/backbone.syphon.js]: TIL `is`

File: spec/javascripts/serialize.spec.js [hunk 2]
```diff
@@ -269,4 +269,40 @@ describe('serializing a form', function() {
...
+      this.view = new this.View();
+      this.view.render();
+
+      // ignore all .doNotSerializeMe elements
+      Backbone.Syphon.ignoredTypes.push('.doNotSerializeMe');
+      this.result = Backbone.Syphon.serialize(this.view);
+    });
+
+    it('should include the specified fields', function() {
+      expect(this.result).to.have.ownProperty('a');
```
Comment (samccone) [inline spec/javascripts/serialize.spec.js]: can we just do a simple .eql instead of two assert blocks?

Comment (rhubarbselleven) [inline spec/javascripts/serialize.spec.js]: Yup sure, I spose the first has been tested to death already

File: src/backbone.syphon.js (hunk 1 above)
Comment (rhubarbselleven) [inline src/backbone.syphon.js]: Yeah I found that in the original PR

> You advanced to level 2

Wooooo

Comment (samccone) [inline src/backbone.syphon.js]: lol

File: spec/javascripts/serialize.spec.js [hunk 3]
```diff
@@ -269,4 +269,35 @@ describe('serializing a form', function() {
...
+      this.view = new this.View();
+      this.view.render();
+
+      // ignore all .doNotSerializeMe elements
+      Backbone.Syphon.ignoredTypes.push('.doNotSerializeMe');
+      this.result = Backbone.Syphon.serialize(this.view);
+    });
+
+    it('should not include fields excluded by selector', function() {
+      expect(this.result).not.to.have.ownProperty('b');
```
Comment (samccone) [inline spec/javascripts/serialize.spec.js]: oh i mean

`expect(this.result).to.eql([....])`

Comment (rhubarbselleven) [inline spec/javascripts/serialize.spec.js]: Ah, gotcha.

Lightbulb :bulb:

Comment (rhubarbselleven) [inline spec/javascripts/serialize.spec.js]: updated :dancers:

**Comment by rhubarbselleven** [comment]
Ugh, hold this. I just wrote a failing testcase for the deserialize side of things

```
expect(this.result).to.eql('myOriginalValue'); // fail !
```

**Comment by rhubarbselleven** [comment]
HA! Was due to a typo

> Rhubarbs learned the power of debugging tests

**Comment by samccone** [comment]
Lgtm. Great work

**Pull request merged**